### PR TITLE
hbase: add remote tso support

### PR DESCRIPTION
**Pull request by disksing**
Note: dsn of hbase is changed to 'zk1,zk2,zk3|tso:port/tbl'.

### CODE REVIEW

File: store/hbase/kv.go [hunk 1]
```diff
@@ -47,8 +49,8 @@ var (
 )
 
 var (
-	// ErrZkInvalid is returned when zookeeper info is invalid.
-	ErrZkInvalid = errors.New("zk info invalid")
+	// ErrDsnInvalid is returned when store dsn is invalid.
+	ErrDsnInvalid = errors.New("dsn invalid")
```
Comment (siddontang) [inline store/hbase/kv.go]: DSN is a initialism.

**Comment by siddontang** [comment]
No test for parsing dsn?

File: store/hbase/kv.go (hunk 1 above)
Comment (qiuyesuifeng) [inline store/hbase/kv.go]: +1, it is better to use as `ErrDSNInvalid`.

Comment (ngaut) [inline store/hbase/kv.go]: s/dsn invalid/invalid dsn/g

Comment (ngaut) [inline store/hbase/kv.go]: s/ErrDSNInvalid/ErrInvalidDSN/g

File: store/hbase/kv.go [hunk 2]
```diff
@@ -108,12 +117,15 @@ func (s *hbaseStore) Close() error {
 }
 
 func (s *hbaseStore) UUID() string {
-	return "hbase." + s.storeName + "." + s.zkInfo
+	return "hbase." + s.storeName + "." + s.dsn
```
Comment (siddontang) [inline store/hbase/kv.go]: fmt.Sprintf() may be better.

File: store/hbase/hbase_test.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+func (t *testHBaseSuite) TestParseDSN(c *C) {
+	zks, oracle, table, err := parseDSN("zk1.com,zk2,192.168.0.1|localhost:1234/tidb")
+	c.Assert(strings.Join(zks, " "), Equals, "zk1.com zk2 192.168.0.1")
+	c.Assert(oracle, Equals, "localhost:1234")
+	c.Assert(table, Equals, "tidb")
+	c.Assert(err, IsNil)
+
+	zks, oracle, table, err = parseDSN("zk1,zk2/tidb")
+	c.Assert(strings.Join(zks, " "), Equals, "zk1 zk2")
```
Comment (siddontang) [inline store/hbase/hbase_test.go]: c.Assert(zks, DeepEquals, []string{"zk1", "zk2"})

**Comment by siddontang** [comment]
Rest LGTM

**Comment by ngaut** [comment]
LGTM

**Pull request merged**